.github/workflows/checks-performance.yaml

```diff
@@ -0,0 +1,92 @@
+---
+#################################################################################
+# Pipeline triggered on every pull request to check performance baseline
+#################################################################################
+name: Performance
+
+on:
+  workflow_call:
+    inputs:
+      source_repo:
+        required: true
+        type: string
+      source_branch:
+        required: true
+        type: string
+
+env:
+  K6_CLUSTER_NODES: local
+  K6_TOPOLOGY_NAME: receiver
+  K6_WORKLOAD_NAME: sanity-check
+  K6_TEST_DURATION: 1
+  K6_ECHO_SERVERS_REPLICAS: 3
+  HOPRD_API_TOKEN: e2e-API-token^^ # only for local testing
+
+concurrency:
+  group: ${{ inputs.source_branch || github.ref_name }}-performance
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    name: Download
+    runs-on: self-hosted-hoprnet-bigger
+    timeout-minutes: 30
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@ec9f2d5744a09debf3a187a3f4f675c53b671911 # v2.13.0
```

Comment:
Please rebase from the latest versions in the master.

Reply:
Done, thanks